servers/rendering/rasterizer_rd/rasterizer_storage_rd.cpp

```diff
@@ -2139,6 +2139,10 @@ void RasterizerStorageRD::mesh_add_surface(RID p_mesh, const RS::SurfaceData &p_
 		mesh->bone_aabbs = p_surface.bone_aabbs;
 		mesh->aabb = p_surface.aabb;
 	} else {
+		// Make sure the bone array is large enough to merge data
+		if (mesh->bone_aabbs.size() < p_surface.bone_aabbs.size())
```

Comment:
Looks good. I think we are trying to more towards always using brackets for ``if`` statements though

Reply:
done

Reply:
Looks like the problem is somewhere else, if the size of bone aabbs is mismatch, this should throw an error and abort the function without adding the surface. Merging wont work because you merge with an empty aabb in position 0,0,0 which is incorrect.